addons/l10n_in_ewaybill_stock_purchase/models/stock_move.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _compute_ewaybill_price_unit(self):
+        super()._compute_ewaybill_price_unit()
+        for line in self.filtered(lambda l: l.purchase_line_id):
+            line.ewaybill_price_unit = line.purchase_line_id.price_unit
+
+    def _compute_tax_ids(self):
+        super()._compute_tax_ids()
+        for line in self.filtered(lambda l: l.purchase_line_id):
+            line.ewaybill_tax_ids = line.purchase_line_id.taxes_id
```

Comment:
This is more tricky (could be a stub method)

Reply:
It is more difficult to not go look on the purchase order to find back the taxes.  (almost impossible)  What we could do is provide a method on stock.move _l10n_in_get_taxes in l10n_in_stock and inherit in a l10n_in_purchase and check if the move has a purchase_line_id.  And then use that method here.  

We need to decide whether we put everything in one module (depending on stock/purchase/sale) or we work with extra methods, because those 3 extra modules are a bit overkill.  @jva-odoo

Reply:
Well I do have second suggestion, though will not match the standards of odoo development but we can simple check if the field exists in the model or not (sale and purchase)